### PR TITLE
fix(error): use text/plain to report errors

### PR DESCRIPTION
**Pull request by tripodsan**
fixes https://github.com/adobe/project-helix/issues/536

https://adobeioruntime.net/api/v1/web/helix/helix-services/content-proxy@ci1739?owner=adobe&repo=spark-website&ref=main&path=/&edit=%3Cscript%3Ealert(document.domain)%3C/script%3E

### CODE REVIEW

**Review by rofe — APPROVED** [review]

**Review by lkrapf — CHANGES_REQUESTED** [review]
Hello @tripodsan

Fix for the reported issue looks ok, but there are more of the same, e.g.: 
https://adobeioruntime.net/api/v1/web/helix/helix-services/content-proxy@v2.html?owner=adobe&repo=spark-website&path=%22%3E%3Cimg%20src%3D///cac%20onerror%3Dalert(document.domain)%3E&edit=foo

You also might want to test your fix with the ".html" extension, otherwise the `text/html` content-type is not added in the first place.

**Comment by tripodsan** [comment]
> You also might want to test your fix with the ".html" extension, otherwise the `text/html` content-type is not added in the first place.

I wonder who sets the content type in this case.... weird.

**Review by lkrapf — CHANGES_REQUESTED** [review]
Setting `text/plain` was the correct fix IMO, and overriding an explicitly set content type based on the extension a very dangerous practice. Your new fix solves the issue of course. +1
As mentioned earlier there's another XSS on line 69 (in the path parameter) - would you like me to create a new issue for that?

**Comment by tripodsan** [comment]
> As mentioned earlier there's another XSS on line 69 (in the path parameter) - would you like me to create a new issue for that?

yes please

**Comment by tripodsan** [comment]
@lkrapf I completely rewrite the error response handling. AFACS, no user input is returned in the response body anymore

File: src/github.js [hunk 1]
```diff
@@ -121,10 +121,7 @@ async function handle(opts) {
       headers,
     });
   }
-  log[utils.logLevelForStatusCode(response.status)](`Unable to fetch ${uri} (${response.status}) from GitHub`);
-  return new Response(body, {
-    status: utils.propagateStatusCode(response.status),
-  });
+  return errorResponse(log, -response.status, `Unable to fetch ${uri} (${response.status}) from GitHub`);
```
Comment (trieloff) [inline src/github.js:124]: is the `-` in `-response.status` intentional?

File: src/google-json.js [hunk 2]
```diff
@@ -63,24 +60,10 @@ async function handleJSON(opts, params) {
...
     }
-
-    log[utils.logLevelForStatusCode(response.status)](`Unable to fetch ${url} (${response.status}) from gdocs2md`);
-    return new Response(body, {
-      status: utils.propagateStatusCode(response.status),
-      headers: {
-        'cache-control': 'max-age=60',
-      },
-    });
+    return errorResponse(log, -response.status,
```
Comment (trieloff) [inline src/google-json.js:63]: Same question.

**Review by trieloff — COMMENTED** [review]
Confused by the `-response.status`

File: src/github.js (hunk 1 above)
Comment (tripodsan) [reply src/github.js:124]: yes... I harmonized the `errorResponse`. there are 2 modes, one that takes the status as is (> 0), and one that treats it as gateway error (<0):

https://github.com/adobe/helix-content-proxy/blob/9b9b44b5f402b00a5ff3b3316490fc489efd63de/src/utils.js#L81-L83

**Review by lkrapf — APPROVED** [review]
:+1:

**Pull request merged**